nbconvert/exporters/templateexporter.py

```diff
@@ -547,14 +546,7 @@ def _template_paths(self, prune=True, root_dirs=None):
             compatibility_dir = os.path.join(root_dir, 'nbconvert', 'templates', 'compatibility')
             paths.append(compatibility_dir)
 
-        additional_paths = self.template_data_paths
-        for path in additional_paths:
-            try:
-                ensure_dir_exists(path, mode=0o700)
-            except OSError:
-                pass
-
-        return paths + self.extra_template_paths + additional_paths
+        return paths + self.extra_template_paths + self.template_data_paths
```

Comment:
> self.template_data_paths

rather than calling `ensure_dir_exists`, should we filter out template data paths that don't exist from the list?

Reply:
Very good point. I added a filtering step, consistent with the code above it, to filter for existing directories if prune is true.